agent/lib_monolog.c

```diff
@@ -397,21 +363,30 @@ NR_PHP_WRAPPER(nr_monolog_logger_addrecord) {
   int api = 0;
   size_t argc = 0;
   char* message = NULL;
+  nr_attributes_t* context_attributes = NULL;
   nrtime_t timestamp = nr_get_time();
 
   /* Values of $message and $timestamp arguments are needed only if log
    * forwarding is enabled so agent will get them conditionally */
   if (nr_txn_log_forwarding_enabled(NRPRG(txn))) {
     argc = nr_php_get_user_func_arg_count(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
-    message = nr_monolog_build_message(argc, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+    message = nr_monolog_build_message(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
```

Comment:
Since `$message` and `$context` are no longer need to be combined, I'd suggest saving two call frames on the stack by replacing call to `nr_monolog_build_message` with call to `nr_monolog_get_message`:
```suggestion
    message = nr_monolog_get_message(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
```
Moreover, `nr_monolog_build_message` function can be removed altogether.

Reply:
Nice catch - I made this suggestion in commit https://github.com/newrelic/newrelic-php-agent/pull/753/commits/aaf9d78eb42c9007ca8cd82769c53c1a02d2260c.